eeps/eep-0072.md

```diff
@@ -0,0 +1,227 @@
+but you can see the usage scattered across a number of modules in the `asn1`
+application).
+
+Usage Syntax Specification
+==========================
+
+This EEP simplifies the above example by
+
+1. Allowing reserved words and variables to be used without quotes for record
+   names, and
+2. Simplifying the definition such that the syntax between record definition
+   and record usage becomes more consistent.
+
+With the changes from this EEP, the above code becomes:
+
+```erlang
+-record('div', {a, b}).
+
+go() ->
+    X = #div{a = 1, b = 2},
+    Y = X#div{a = something_else},
+    Z = Y#div.a,
+    ...
+```
+
+Definition Syntax Motivation
+============================
+
+While the updated example in the usage syntax specification makes the *using*
+of records cleaner, there remains one more inconsistency that can also be
+relatively easily solved. That is the record definition still also needing to
+quote record name, as the example above demonstrates (repeated here for
+convenience):
+
+```erlang
+-record('div', {a, b}).
+
+go() ->
+    X = #div{a = 1, b = 2},
+    Y = X#div{a = something_else},
+    Z = Y#div.a,
+    ...
+```
+
+So whereas the record definition needs to be thought of as `'div'`, the record
+usage no longer requires the quoted term 'div', which could certainly lead an
+Erlang beginner to wonder why 'div' needs to be quoted in the definition while
+other atom-looking terms don't.
+
+Definition Syntax Specification
+===============================
+
+Conveniently, there is a rather easy solution, and that's to
+allow the record usage syntax to also be used as the record definition.
+
+This EEP also then also adds a new record definition syntax, improving the
+symmetry between general record usage and record definition.
+
+The above example can fully then look like the following:
+
+```erlang
+-record #div{a, b}.
+
+go() ->
+    X = #div{a = 1, b = 2},
+    Y = X#div{a = something_else},
+    Z = Y#div.a,
+    ...
+```
+
+Implementation
+==============
+
+To update the syntax for using records, we can safely augment the parser to
+change `# atom` to `# record_name` and define `record_name` to be either
+`atom`, `var`, or `reserved_word`.
+
+To update the record definition syntax, we can simply add a few new
+modifications to the `attribute` Nonterminal.
+
+Backwards Compatibility
+=======================
+
+As this EEP only adds new syntax, the vast majority existing codebases will
+still work, with the possible exception of AST/code analysis tools that are
+analyzing code using the new syntax.
+
+Syntax highlighting and code completion tools may need to be updated to support
+the new syntax if your code uses the new syntax rules.
+
+Broader Concerns and Points of Discussion
+=========================================
+
+While the new definition syntax creates some degree of symmetry around record
+usage, there do remain some concerns and inconsistencies:
```

Comment:
```suggestion
usage, perfect symmetry is impossible to achieve, since a record can always
be handled as the atom tagged tuple it actually is. The question is where
to draw the line where the record's true nature shows, and how hard we
should try to hide it. These are remaining concerns and inconsistencies: